chat-commands.js

```diff
@@ -2377,19 +2377,22 @@ exports.commands = {
 	hidealtstext: 'hidetext',
 	hidetext: function (target, room, user, connection, cmd) {
 		if (!target) return this.parse('/help hidetext');
+		if (!room.log.hasUsername(target)) return false;
 
 		this.splitTarget(target);
 		let targetUser = this.targetUser;
 		let name = this.targetUsername;
-		if (!targetUser) return this.errorReply("User '" + name + "' not found.");
-		let userid = targetUser.getLastId();
+		if (!targetUser && !user.can('lock')) return this.errorReply("User '" + name + "' not found.");
+		let userid = targetUser ? targetUser.getLastId() : "";
```

Comment:
`''`

(CONTRIBUTING.md does talk about what string delimiters you should use in what situation, for the record)

Reply:
Also I hope you realize that your code doesn't actually work. At no point do you define `userid` correctly for offline users.